### ISSUE TITLE
debugging "Resource temporarily unavailable"

### ISSUE THREAD

**Issue by tatraian** [lxc/lxd#3764]
Hello,

I use lxd to develop a program. Basically my application is built and runs in a CentOS 6 container that runs on my 17.04 Kubuntu. Sometimes (not always) I get error when I start this application (it contains several processes and inside the processes there are several threads). I always get this kind of error:
```
cant create thread, errno=11, Resource temporarily unavailable
```

I tried to figure out what limit breaks the start but I don't find any log. I checked the next things:
 - `journalctl` on host machine
 - `dmesg` on host machine
 - `systemctl status lxd` on host machine
 - files in `/var/log/lxd` on host machine
 - `/var/log/messages` inside container

So my question is: where are logged these kinds of errors? Or what have to switch on to see them?

**Comment by brauner** [comment]
Can you please show the output of:

- `cat /proc/sys/kernel/threads-max`
- `cat /proc/sys/kernel/pid_max`
- `ulimit -u`
- `ps auxH | wc -l`

I suspect that your process limit is too low

**Comment by tatraian** [comment]
The limits are the next on the host machine as root:
```
root@SGBPPC098:~# cat /proc/sys/kernel/threads-max 
63007
root@SGBPPC098:~# cat /proc/sys/kernel/pid_max 
32768
root@SGBPPC098:~# ulimit -u
31503
root@SGBPPC098:~# ps auxH | wc -l
838
root@SGBPPC098:~# ps auxH | wc -l
1326
root@SGBPPC098:~# ps auxH | wc -l
867
```

First `ps` output is the system in idle. Second when application has been started, third when a youtube playing is running and in that case the application cannot start.
And the same information from my container:
```
Antal.Tatrai@xion:~$ cat /proc/sys/kernel/threads-max 
63007
Antal.Tatrai@xion:~$ cat /proc/sys/kernel/pid_max 
32768
Antal.Tatrai@xion:~$ ulimit -u
1024
Antal.Tatrai@xion:~$ ps auxH | wc -l
59
```

**Comment by stgraber** [comment]
For your first question, that particular error comes straight from the kernel and unless it triggers a kprintf where you'd get an entry in "dmesg", that's about as much information as you'll get from the kernel...

Does running the container privileged somehow fixes this?

**Comment by tatraian** [comment]
Can I change the type of my container with command:
```
lxc config set my_container security.privileged true
```

Will this cause problem of file ownerships?

**Comment by stgraber** [comment]
You'll have to restart the container after you change that property. LXD will re-map all the container files for you (well, all of those that are part of the container's rootfs anyway).

**Comment by tatraian** [comment]
Thanks! I turned my container to privileged and that doesn't solved the problem BUT I found that my user's process limit remains 1024, so I figured out that this limitation must comes from Cent OS setup. I found file:
`/etc/security/limits.d/90-nproc.conf` that contains the next:
```
*          soft    nproc     1024
root       soft    nproc     unlimited
```

So I modified this threshold to unlimited and that solved the problem. I'm not sure but I think this is because my container user is mapped to my real user. In this case the kernel dropped my new process requests in container where the user limit was only 1024 but never dropped in my real user's requests that has limit 31503. But my real users number of processes were counted in case of my container user too.

Thanks your very-very fast help!
